dlib/CMakeLists.txt

```diff
@@ -385,7 +385,12 @@ if (NOT TARGET dlib)
                external/zlib/uncompr.c
                external/zlib/zutil.c
                )
-            if (NEON)
+
+            include(cmake_utils/use_arm_neon.cmake)
+            include(CheckCXXCompilerFlag)
+            CHECK_CXX_COMPILER_FLAG("-mfpu=neon" NEON_DEFINED)
```

Comment:
It seems like you don't need the -fpu=neon check.  If the other check passes then we know we are building in ARM mode and can compile the NEON files right?  

Also, just because someone has a compiler that is capable of cross compiling to ARM doesn't mean they want to use it.  If they don't actually give the compiler -mfpu=neon then the build will fail, but this NEON_DEFINED check will pull in the libpng NEON files regardless of the presence of -mfpu=neon in the actual compiler settings.